simulator/src/main/java/com/hazelcast/simulator/worker/RunStrategy.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.simulator.worker;
+
+import java.util.concurrent.Callable;
+
+/**
+ * A RunStrategy encapsulates the logic for the 3 different types of test running approaches:
+ * <ol>
+ * <li>{@link com.hazelcast.simulator.test.annotations.Run}</li>
+ * <li>{@link com.hazelcast.simulator.test.annotations.RunWithWorker}</li>
+ * <li>{@link com.hazelcast.simulator.test.annotations.TimeStep}</li>
+ * </ol>
+ */
+public abstract class RunStrategy implements Callable {
```

Comment:
The reason for `Callable` is the `throws Exception` which the `Runnable` doesn't have, right? I guess the return value is always ignored though.